xpod.go

```diff
@@ -8,99 +9,146 @@ import (
 	"log"
 	"net/http"
 	"os"
+	"time"
 )
 
 var t_site = "https://xray.fm"
 var t_shows = t_site + "/shows"
 
-func fetch(url string) (*goquery.Document, error) {
+// fetch url, returning either a parsed goquery Document or an error.
+func fetch(url string) (*http.Response, *goquery.Document, error) {
 	var doc *goquery.Document
 
-	res, err := http.Get(url)
+	resp, err := http.Get(url)
 	if err != nil {
-		return doc, errors.Wrapf(err, "Failed to fetch: `%s'", url)
+		return nil, nil, errors.Wrapf(err, "Failed to fetch: `%s'", url)
 	}
-	defer res.Body.Close()
-	if res.StatusCode != 200 {
-		return doc, errors.Errorf("`%s' status code error: %d %s", url, res.StatusCode, res.Status)
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, nil, errors.Errorf("`%s' error: %s", url, resp.Status)
 	}
 
 	// Load the document
-	doc, err = goquery.NewDocumentFromReader(res.Body)
+	doc, err = goquery.NewDocumentFromReader(resp.Body)
 	if err != nil {
-		return doc, errors.Wrapf(err, "Failed to parse `%s'", url)
+		return nil, nil, errors.Wrapf(err, "Failed to parse `%s'", url)
 	}
-	return doc, nil
-}
 
-func make_feed(url string) (feeds.Feed, error) {
-	var feed feeds.Feed
+	return resp, doc, nil
+}
 
-	doc, err := fetch(url)
+// Given a url to an xray show, create a Feed from it
+func make_feed(url string) (*feeds.Feed, error) {
```

Comment:
```suggestion
func makeFeed(url string) (*feeds.Feed, error) {
```

Idiomatic Go is `camelCase`, not `snake_case`.